src/screens/Exchange/Exchange/Exchange.js

```diff
@@ -73,46 +79,72 @@ function Exchange() {
+  const { address: nativeChainAssetAddress } = nativeAssetPerChain[initialChain];
+  const initialFromAddress: string = navigation.getParam('fromAssetAddress') || nativeChainAssetAddress;
+  const initialToAddress: string = navigation.getParam('toAssetAddress') || getPlrAddressForChain(initialChain);
 
-  const [fromSymbol, setFromSymbol] = React.useState(initialFromSymbol);
-  const [toSymbol, setToSymbol] = React.useState(initialToSymbol);
+  const [chain, setChain] = React.useState(initialChain);
+  const [fromAddress, setFromAddress] = React.useState(initialFromAddress);
+  const [toAddress, setToAddress] = React.useState(initialToAddress);
 
   const [rawFromAmount, setFromAmount] = React.useState('');
   const [fromAmount]: [string] = useDebounce(rawFromAmount, 500);
 
-  const chainConfig = useChainConfig(CHAIN.ETHEREUM);
-  const ethereumSupportedAssets = useChainSupportedAssets(CHAIN.ETHEREUM);
+  const chainConfig = useChainConfig(chain);
+
+  const assetsPerChain = useRootSelector(accountAssetsPerChainSelector);
+  const supportedAssetsPerChain = useSupportedAssetsPerChain();
+  const ratesPerChain = useRatesPerChain();
+  const accountBalances = useRootSelector(accountAssetsBalancesSelector);
+  const walletBalancesPerChain = getChainWalletAssetsBalances(accountBalances);
 
   const fromOptions = React.useMemo(
-    () => getExchangeFromAssetOptions(assets, ethereumSupportedAssets, balances, fiatCurrency, ethereumRates),
-    [assets, ethereumSupportedAssets, balances, fiatCurrency, ethereumRates],
+    () => supportedExchangeChains.reduce((multiChainOptions, supportedChain) => {
+      const chainOptions = getExchangeFromAssetOptions(
+        assetsPerChain,
+        supportedAssetsPerChain,
+        walletBalancesPerChain,
+        fiatCurrency,
+        ratesPerChain,
+        supportedChain,
+      );
+      return [...multiChainOptions, ...chainOptions];
+    }, []),
+    [assetsPerChain, supportedAssetsPerChain, walletBalancesPerChain, fiatCurrency, ratesPerChain],
   );
 
-  const toOptions = React.useMemo(() => getExchangeToAssetOptions(
-    ethereumSupportedAssets,
-    balances,
-    fiatCurrency,
-    ethereumRates,
-  ), [
-    ethereumSupportedAssets,
-    balances,
-    fiatCurrency,
-    ethereumRates,
-  ]);
-
-  const fromAsset = React.useMemo(() => fromOptions.find((a) => a.symbol === fromSymbol), [fromOptions, fromSymbol]);
-  const toAsset = React.useMemo(() => toOptions.find((a) => a.symbol === toSymbol), [toOptions, toSymbol]);
+  const toOptions = React.useMemo(
+    () => supportedExchangeChains.reduce((multiChainOptions, supportedChain) => {
+      const chainOptions = getExchangeToAssetOptions(
+        supportedAssetsPerChain,
+        walletBalancesPerChain,
+        fiatCurrency,
+        ratesPerChain,
+        supportedChain,
+      );
+      return [...multiChainOptions, ...chainOptions];
+    }, []),
+    [supportedAssetsPerChain, walletBalancesPerChain, fiatCurrency, ratesPerChain],
```

Comment:
AFAIR the original logic, picking the from assets restricted the available to asset options, so you may want to check that with Dmitry.

Reply:
Per what I discussed with Dmitry before we allow to select either `from` or `to` asset on any chain and chain on the screen changed accordingly, but still allowing to choose assets from any chain.